### ISSUE TITLE
Federation fails when filtering by multiple tags

### ISSUE THREAD

**Issue by fmbiete** [VictoriaMetrics/VictoriaMetrics#1001]
**Describe the bug**
After migrating from Prometheus to VictoriaMetrics we are seeing federation errors when remote data sources request metrics filtering by multiple tags. For example; a remote Prometheus request metrics for 3 different servers.

We get the following in VictoriaMetrics log
```
2021-01-08T00:39:25.363Z        warn    /home/builder/rpm/BUILD/VictoriaMetrics-pmm-6401-v1.50.2/app/vmselect/main.go:241       remoteAddr: "127.0.0.1:43878", X-Forwarded-For: "10.48.144.41"; error in "/federate": cannot fetch data for "MinTimestamp=2021-01-08 00:34:24 +0000 UTC, MaxTimestamp=2021-01-08 00:39:24 +0000 UTC, TagFilters=[\n{Key=\"node_name\", Value=\"serverA\", IsNegative: false, IsRegexp: false}\n{Key=\"node_name\", Value=\"serverB\", IsNegative: false, IsRegexp: false}\n{Key=\"node_name\", Value=\"serverC\", IsNegative: false, IsRegexp: false}\n]": error when searching in Prometheus data: unexpected len(sq.TagFilters); got 3; want 1
```

**To Reproduce**
Use the following Prometheus configuration for the remote datasource (the one requesting data through federation):
```
scrape_configs:
  - job_name: 'federate'
    scrape_interval: 60s

    honor_labels: true
    metrics_path: '/federate'

    scheme: https
    tls_config:
      insecure_skip_verify: true

    params:
      'match[]':
        - '{node_name="serverA"}'
        - '{node_name="serverB"}'
        - '{node_name="serverC"}'

    static_configs:
      - targets:
         - 'victoriametrics.domain.com:443'
```

**Expected behavior**
Data for the 3 servers should be returned and no warning/error should be logged

**Version**
The line returned when passing `--version` command line flag to binary. For example:
```
$ victoriametrics --version
victoria-metrics-20201224-155349-pmm-6401-v1.50.2
```

**Comment by valyala** [comment]
This issue should be fixed in the commit f5e70f0ab914198461d5b3ba1ac710e7e244bba2 . This commit will be included in the next release.

**Comment by freeseacher** [comment]
I have this issue too. building vm from c97681b45c808537074da122cdd1256c131768d6 and vm-select continues work for at least an hour

**Comment by valyala** [comment]
The issue must be fixed in [v1.52.0 release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing it as done.
